### ISSUE TITLE
[feature]: Module based chunking along with route based chunking.

### ISSUE THREAD

**Issue by revanth0212** [magento/pwa-studio#1804]
**Is your feature request related to a problem? Please describe.**
Related to #1673 #1765 #1783 

**Describe the solution you'd like**
We can chunk the app based on its availability requirement in the app. For instance, the left drawer is needed in all the pages. We can separate out the left drawer or combine that with app shell and sever it as a bundle of its own so it should be cached on the UI and reused until something changes. Also, we can server render these bundles for better performance (if possible).

**Please let us know what packages this feature is in regards to:**
- [x] `venia-concept`
- [x] `venia-ui`
- [x] `pwa-buildpack`
- [x] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

**Comment by revanth0212** [comment]
This needs more research. Need to check on what Root Components Chunking is and how is that is related to what this issue addresses.

**Comment by supernova-at** [comment]
#1765 introduced route-based code splitting.

Edit: Oh I see you've already linked it :P :)

**Comment by revanth0212** [comment]
> #1765 introduced route-based code splitting.
 
Yeah, this adds chunking on top of what #1765 offers. #1765 already gives us a boost with a smaller `client.js` which can be broken up into even smaller chunks using module-based chunking.

**Comment by cherdman** [comment]
https://jira.corp.magento.com/browse/PWA-72

**Comment by revanth0212** [comment]
This has a high workload. Removing from the Performance Milestone for now. Will revisit in the future.

**Comment by glo82145** [comment]
As this is a very old open issue and no update from so long hence we are closing this
